actors/builtin/miner/miner_test.go

```diff
@@ -865,6 +865,173 @@ func TestCommitments(t *testing.T) {
 		rt.Verify()
 	})
 
+	t.Run("try to upgrade committed capacity sector twice", func(t *testing.T) {
+		actor := newHarness(t, periodOffset)
+		rt := builderForHarness(actor).
+			WithBalance(bigBalance, big.Zero()).
+			Build(t)
+		actor.constructAndVerify(rt)
+
+		// Move the current epoch forward so that the first deadline is a stable candidate for both sectors
+		rt.SetEpoch(periodOffset + miner.WPoStChallengeWindow)
+
+		// Commit a sector to upgrade
+		// Use the max sector number to make sure everything works.
+		oldSector := actor.commitAndProveSector(rt, abi.MaxSectorNumber, defaultSectorExpiration, nil)
+
+		// advance cron to activate power.
+		advanceAndSubmitPoSts(rt, actor, oldSector)
+
+		st := getState(rt)
+		dlIdx, partIdx, err := st.FindSector(rt.AdtStore(), oldSector.SectorNumber)
+		require.NoError(t, err)
+
+		// Reduce the epoch reward so that a new sector's initial pledge would otherwise be lesser.
+		actor.epochReward = big.Div(actor.epochReward, big.NewInt(2))
+		actor.epochRewardSmooth = smoothing.TestingConstantEstimate(actor.epochReward)
+
+		challengeEpoch := rt.Epoch() - 1
+
+		// Upgrade 1
+
+		upgradeParams1 := actor.makePreCommit(200, challengeEpoch, oldSector.Expiration, []abi.DealID{1})
+		upgradeParams1.ReplaceCapacity = true
+		upgradeParams1.ReplaceSectorDeadline = dlIdx
+		upgradeParams1.ReplaceSectorPartition = partIdx
+		upgradeParams1.ReplaceSectorNumber = oldSector.SectorNumber
+		upgrade1 := actor.preCommitSector(rt, upgradeParams1)
+
+		// Check new pre-commit in state
+		assert.True(t, upgrade1.Info.ReplaceCapacity)
+		assert.Equal(t, upgradeParams1.ReplaceSectorNumber, upgrade1.Info.ReplaceSectorNumber)
+		// Require new sector's pledge to be at least that of the old sector.
+		assert.Equal(t, oldSector.InitialPledge, upgrade1.PreCommitDeposit)
```

Comment:
Not your bug but the comment is incorrect and its hiding that something is wrong.  The assertion captures the correct invariant, new sectors PCD is at least old initial pledge.  Looking at the code I don't think we enforce this requirement on the upgrade's IP.  I think this is different than the expected behavior.  I will file an issue.

Reply:
I _think_ it's implicit because the new sector will live longer, but I'm not sure as pledge requirements change over time...

Reply:
Discussion in #1021